test/Test/Pos/UtilSpec.hs

```diff
@@ -112,3 +153,38 @@ verifyDiffMapIsSmaller (SmallHashMap hm1) (SmallHashMap hm2) =
         sumValSizes = sum . fmap length
     -- (p || q) <=> ((not p) => q)
     in (null commonKey) || (sumValSizes hm1 > sumValSizes diffMap)
+
+toSingletonN
+    :: (Arbitrary a, Show a, Eq a)
+    => a
+    -> Expectation
+toSingletonN x = one x `shouldBe` (NewestFirst @[] . one $ x)
+
+toSingletonO
+    :: (Arbitrary a, Show a, Eq a)
+    => a
+    -> Expectation
+toSingletonO x = one x `shouldBe` (OldestFirst @[] . one $ x)
```

Comment:
I guess `toSingleton0` and `toSingletonN` can be merged and abstracted over constructors. I also don't like indentation.